src/core/mormot.core.os.pas

```diff
@@ -5871,6 +5871,8 @@ procedure TSynLocker.RWUnLock(context: TRWLockContext);
 
 procedure TSynLocker.Lock;
 begin
+  if fSection.LockCount<0 then

```

Comment:
This is very specific to Windows, and won't compile with Linux/POSIX.
I don't understand what you want to achieve.